packages/react-router-dom/modules/BrowserPrompt.js

```diff
@@ -0,0 +1,80 @@
+import React, { PropTypes } from 'react'
+import { Prompt } from 'react-router'
+
+const BEFOREUNLOAD = "beforeunload";
+
+var UnloadComponent = ComposedComponent => class extends React.Component {
```

Comment:
Can we please just use a regular component here? Higher-order components are unnecessary and just make stuff harder to read.